src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependencyNode.cs

```diff
@@ -199,7 +201,7 @@ public void AddChild(IDependencyNode childNode)
         {
             Requires.NotNull(childNode, nameof(childNode));
 
-            lock(_children)
+            lock(_childrenLock)
             {
                 _children = _children.Add(childNode);
```

Comment:
And, in general, whether the lock is really needed for such operations as assignment and returns for variables of reference type?